### DISCUSSION TITLE
The coupled heat source from openmc cannot act on the temperature field

### DISCUSSION THREAD

**Discussion by LeonDong1998** [idaholab/moose#28319] [Q&A Modules: General]
Hello everyone!

I have installed and deployed cardinal-moose-openmc and run it successfully. When running the example, I found a problem. The temperature field in moose solid seems to be affected by the power value of 1000, and the calculated heat source (power 1e7) does not act on the temperature field.
The heat source calculated by openmc exists in the form of auxvariables, but the heatsource kernel seems to only support equations and value. How to set the auxiliary variables as internal heat sources?

Thanks in advance!

**Comment by GiudGiud** [comment]
hello

you can use this kernel to input a force term (here a heat source) from a variable
https://mooseframework.inl.gov/source/kernels/CoupledForce.html
